app/models/submission.rb

```diff
@@ -663,6 +941,10 @@ def submission_history
       last_submitted_at = nil
       self.versions.sort_by(&:created_at).reverse_each do |version|
         model = version.model
+        # since vericite_data is a function, make sure you are cloning the most recent vericite_data_hash
+        if self.vericiteable?
+          model.turnitin_data = self.vericite_data
```

Comment:
This could side effect into network calls when bringing up submission history.  This isn't a good idea here at scale.